firmware/application/ui/ui_alphanum.cpp

```diff
@@ -65,10 +69,19 @@ AlphanumView::AlphanumView(
         set_mode(mode + 1);
     };
 
+    button_delete.on_select = [this](Button&) {
+        char_delete();
+    };
+
     field_raw.set_value('0');
     field_raw.on_select = [this](NumberField&) {
         char_add(field_raw.value());
     };
+
+    // make text_raw_to_char widget display the char value from field_raw
+    field_raw.on_change = [this](auto) {
```

Comment:
See comment in header. It would be best if didn't need to reallocate the "AKA:" every time you turn the dial. Here you're allocating quite a few new string instances every time.

Best case would be if you didn't need to allocate at all here. You'd have to handle paint and draw it yourself though. Probably not really worth the extra code needed though.

Reply:
It's not about crashes/slow, it's just about unnecessary work that may lead to fragmentation.
Ideally you could use a Label control for "Raw:" and "AKA:" because "Text" is meant to be changed where Labels are not.

Reply:
Sure, i will use label widget for those two const text. However, i'm afraid I have to add one more widget if so, cuz Label can't seems crossing lines, and i don't wanna touch much for it, is it ok to you?

Reply:
Label lets you place multiple Labels.

Reply:
```
    Labels labels{
        { { 0, 20 * 8 + 4 }, "Title:", Color::light_grey() },
        {{0, 180}, "Key:", Color::light_grey()}};```

For example ^^

Reply:
OHHHH thank you for teaching me this! will use it XD

Reply:
Thank you, resolved.